sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -42,5 +45,99 @@ public virtual Response<Completions> GetCompletions(string deploymentId, string
             completionsOptions.Prompt.Add(prompt);
             return GetCompletions(deploymentId, completionsOptions, cancellationToken);
         }
+
+        public virtual Response<StreamingCompletions> GetCompletionsStreaming(
+            string deploymentId,
+            CompletionsOptions completionsOptions,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(deploymentId, nameof(deploymentId));
+            Argument.AssertNotNull(completionsOptions, nameof(completionsOptions));
+
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope(
+                "OpenAIClient.GetCompletionsStreaming");
+            scope.Start();
+
+            RequestContext context = FromCancellationToken(cancellationToken);
+
+            RequestContent nonStreamingContent = completionsOptions.ToRequestContent();
+            RequestContent streamingContent = GetStreamingEnabledRequestContent(nonStreamingContent);
+
+            try
+            {
+                HttpMessage message = CreateGetCompletionsRequest(deploymentId, streamingContent, context);
+                message.BufferResponse = false;
```

Comment:
You should be able to use `x-csharp-buffer-response` to have AutoRest generate this for you - if your version of the Cadl tooling allow a decorator like `@extension("x-csharp-buffer-response", false)`.

Reply:
That'll be a *much* nicer way of achieving this and cut down on the custom code significantly. I believe we're not quite there yet with the cadl, but should follow up ASAP to converge. I'll move forward with it as-is here for expediency's sake.